registry/remote/auth/scope.go

```diff
@@ -54,6 +58,28 @@ func ScopeRepository(repository string, actions ...string) string {
 	}, ":")
 }
 
+// AppendScopeHints returns a new context containing scope hints for the auth
+// client to fetch bearer tokens with the given actions on the repository.
+// If called multiple times, the new scopes will be appended to the existing
+// scopes. The resulted scopes are de-duplicated.
+//
+// For example, uploading blob to the repository "hello-world" does HEAD request
+// first then POST and PUT. The HEAD request will return a challenge for scope
+// `repository:hello-world:pull`, and the auth client will fetch a token for
+// that challenge. Later, the POST request will return a challenge for scope
+// `repository:hello-world:push`, and the auth client will fetch a token for
+// that challenge again. By invoking AppendScopeHints with the actions
+// [ActionPull] and [ActionPush] for the repository `hello-world`,
+// the auth client with cache is hinted to fetch a token via a single token
+// fetch request for all the HEAD, POST, PUT requests.
+func AppendScopeHints(ctx context.Context, ref registry.Reference, actions ...string) context.Context {
```

Comment:
It's hard to name this method...

Reply:
Yeah I agree. Let's call it `AppendRepositoryScope`.
// Just that I liked the idea of "hint"😄.

Reply:
I even thought of "HintRepositoryScope"...